include/emul.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_EMUL_H_
+#define ZEPHYR_INCLUDE_EMUL_H_
+
+/**
+ * @brief Emulators used to test drivers and higher-level code that uses them
+ * @defgroup io_emulators Emulator interface
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+struct device;
+struct emul;
+
+/**
+ * Structure uniquely identifying a device to be emulated
+ *
+ * Currently this uses the device node label, but that will go away by 2.5.
+ */
+struct emul_link {
```

Comment:
This should be `emul_link_for_bus` since it's what gets processed by `emul_list_for_bus`.  We don't know what we'll want for emulator identifiers for things that are not bus children.

Reply:
done